### DISCUSSION TITLE
Question regarding static graph

### DISCUSSION THREAD

**Discussion by himoto** [biomass-dev/biomass#152] [Q&A] [answered]
Hi @formersbach,
In the example of [NFkB model](https://biomass-core.readthedocs.io/en/latest/graph_visualization.html#graph-generation-and-static-image), Can I change the shape or color for each species based on its status, e.g., mRNA and proteins to be enclosed with circles and squares, respectively?

**Comment by formersbach** [comment, marked as answer]
Yes it's possible, although somewhat tedious. 
You will have to define the shape of each node individually. You can get a list of nodes using:
```python
model = create_model('path.to.model')
model.to_graph()
G = model.graph
G.nodes
```

Then each node can be edited using:

```python
n = G.get_node('nodename')
n.attr["shape"] = "triangle"

#Equivalent to using "-Nshape=paralellogram" in static_plot() :
for node in G.nodes:
    n = G.get_node[node]
    n.attr["shape"] = "paralellogram"
```
Detailed available attributes can be found in [graphviz doc](https://graphviz.org/pdf/dot.1.pdf) and [pygraphviz doc](https://pygraphviz.github.io/documentation/stable/tutorial.html).
Hope this answers your question!

**Reply by himoto** [reply]
Thank you for your reply!
In the example of NF-kB model,
```python
model.to_graph()
G = model.graph
for sname in model.species:
    n = G.get_node(sname)
    if sname.startswith("m") or sname.startswith("Rna"):
        n.attr["shape"] = "triangle"
    else:
        n.attr["shape"] = "paralellogram"
```
After running the code above, I got the expected one!
Note that when I run the code:
```python
for node in G.nodes:
    ...
```
I got `TypeError: 'method' object is not iterable`


Thank you
